src/exceptions.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+use rquickjs::{
+    atom::PredefinedAtom,
+    function::Opt,
+    module::{Declarations, Exports, ModuleDef},
+    Class, Ctx, Result,
+};
+
+use crate::module::export_default;
+
+#[derive(Clone, Default)]
+#[rquickjs::class]
+#[derive(rquickjs::class::Trace)]
+pub struct DOMException {
+    message: String,
+    name: String,
+}
+
+#[rquickjs::methods]
+impl<'js> DOMException {
```

Comment:
I think we might need to inherit from `Error` in order for stack traces to be populated. Please verify that stacks are populated if throwing.

Reply:
Sorry @richarddavison , this effort seems to have been too difficult for me. I have tried many times, but I just can't get the implementation to combine the `error trait` and `rquickjs` and how to return it to the Javascript side as an Exception. Can you please assist me with the implementation?

In trying to incorporate `AbortError` and `TimeoutError`, I am not even sure if the `DOMException` structure is correct in this form.

Reply:
Hi i think it’s fine to do what you’re already doing. You can test by just running `throw new DOMException()`

Reply:
Hi @richarddavison.
As for what I can do, I just added the missing test cases.

Reply:
We decided to go back to the drawing board and inherit the `Error` pointed out. We also added a definition of `fmt()`, which is mandatory to implement.
I ported the code from the `toString()` method, but since the return value is not a `String`, calling it from JavaScript seems to return an `object`, not a `string`. This is not important at this time, so I replaced it with a `todo!`.

Is the next step to allow DOMException() to hold Stack?

Reply:
I think actually it would be simpler to do this:
```rust
#[rquickjs::class]
#[derive(rquickjs::class::Trace)]
struct DOMException {
    message: String,
    name: String,
    stack: String,
}

#[rquickjs::methods]
impl DOMException {
    #[qjs(constructor)]
    fn new<'js>(ctx: Ctx<'js>, message: String, name: String) -> Result<Self> {
        let error_ctor: Constructor = ctx.globals().get(PredefinedAtom::Error)?;
        let new: Object = error_ctor.construct((message.clone(),))?;

        Ok(Self {
            name,
            message,
            stack: new.get::<_, String>(PredefinedAtom::Stack)?,
        })
    }

    #[qjs(get)]
    fn name(&self) -> String {
        self.name.clone()
    }

    #[qjs(get)]
    fn stack(&self) -> String {
        self.stack.clone()
    }

    #[qjs(get)]
    fn message(&self) -> String {
        self.message.clone()
    }
}
```
And then set inheritance by setting the prototype, where you define the class (in init)
```rust
Class::<DOMException>::define(&globals)?;
let dom_ex_proto = Class::<DOMException2>::prototype(ctx.clone()).unwrap();
let error_ctor: Object = globals.get(PredefinedAtom::Error)?;
let error_proto = error_ctor.get_prototype();
dom_ex_proto.set_prototype(error_proto.as_ref())?;
```

Reply:
Thanks for the detailed advice.
We have modified the implementation based on what has been pointed out to us.
In addition, we were able to pass all the test cases we had prepared.